deno/lib/__tests__/refine.test.ts

```diff
@@ -217,3 +217,25 @@ test("fatal superRefine", () => {
   expect(result.success).toEqual(false);
   if (!result.success) expect(result.error.issues.length).toEqual(1);
 });
+
+test("fatal refine", () => {
+  const Strings = z
+    .string()
+    .refine((val) => val === "", {
+      message: "foo",
+      fatal: true,
+    })
+    .superRefine((val, ctx) => {
```

Comment:
@DanielBreiner Would it make sense to add also a test that flexes the `fatal` option stopping processing before the next `refine`/`superRefine`? 

As it is, you're testing the second `superRefine`'s message, but we're not testing that the `fatal` option actually stops processing. Something like your original issue use case would work great IMO!

```ts
  const testSchema = z
    .object({ test: z.literal(true) })
    .nullable()
    .refine((val) => val !== null, {
      message: "foo",
      fatal: true, // errors without this 
    })
    .superRefine((val, ctx) => {
      if ((val as any).test) {
        console.log("actually made it here");
        ctx.addIssue({
          code: z.ZodIssueCode.custom,
          message: "bar",
        });
      }
    });

  const result = testSchema.safeParse(null);
```

Reply:
Good idea. Added.